packages/snjs/lib/services/api/http_service.ts

```diff
@@ -125,7 +125,9 @@ export class SNHttpService extends PureService {
     try {
       if (httpStatus !== StatusCode.HttpStatusNoContent) {
         const body = JSON.parse(request.responseText);
-        response.object = body;
+        if (!body.data) {
+          response.data = body;
+        }
         Object.assign(response, body);
```

Comment:
So `response` might contain `body` in its `data` field (via the added `if`) and also all of `body`'s properties at the same time. Is that ok?

Reply:
`body` is the whole request body, not a specific key.
Under the future api, response bodies will look closer to  be contained in a root-level `data` key (there will also be a `meta` key, to move closer to a [JSON:API](https://jsonapi.org/)-complient API shape). This code is here to handle legacy responses that may not have a data key, allowing consumers of the httpService to treat everything the same way regardless of the response shape.
Your remark makes me realize that it should be explained somehow!

Reply:
I've pushed an update with a comment, let me know if that makes things clearer :)

Reply:
I see. So we duplicate the data for compatibility.